lib/web_ui/dev/test_runner.dart

```diff
@@ -49,13 +49,14 @@ class TestsCommand extends Command<bool> {
 
     _copyAhemFontIntoWebUi();
     await _buildHostPage();
-    await _buildTests();
 
     final List<FilePath> targets =
         this.targets.map((t) => FilePath.fromCwd(t)).toList();
     if (targets.isEmpty) {
+      await _buildTests();
       await _runAllTests();
     } else {
+      await _buildTests(targets: targets);
```

Comment:
Let's call them `_buildAllTests` and `_buildTargetTests` to keep the naming consistent?

Reply:
Actually, I'm being silly. `_buildTests(targets: targets)` works in both cases. The behavior of `-t` w.r.t. to empty list is the same as `--build-filter`.

Deduped.